### ISSUE TITLE
colony 449 capacity seen from other account

### ISSUE THREAD

**Issue by marcgfx** [Skratti/EmpiresInSpace#158]
capacity is 8000 for marcgfx
capacity is 5000 for Marvin (can not transfer goods)

**Comment by FlexiGit** [comment]
This is a general problem and makes trading very inconvenient. You can not just drop your wares on the planet and fly on. Only the planet owner can do the trade. So you have to wait form him to log on.

My guess is, that warehouses are only calculated in the client if you can see them (you are the planet owner).

**Comment by Skratti** [comment]
Yes, capacity is calculated by summing up the buidling cargo capacities.

Building information is not transfered to other players (because colony information is always send on startup, and transferring all buildings seemded a bit overkill).

Workaround: calculdate that value on server side and cach the result. cache has to be cleared if a building is built/razed/deactivated.

**Comment by Skratti** [comment]
Mostly implemented in the dev environment and will be pushed to the productive system asap.

Missing: if a user builds/razes/deactivates or activates a building, other users who are online at the same time are not informed, and thus have incorrect capacity informations.

Solution: Use WebSocket to push colonyId to all online users, so that they fetch the colony info again (if they do now of that colony already).
